src/index.js

```diff
@@ -12,171 +12,179 @@ const dirScanner = require('./dirScanner')
-    const {scanStudies} = defaults;
-    const verbose = hasArg('-v', '--verbose', false, 'Write verbose output')
-
-    const isHelp = hasArg('-h', '--help',false, 'Print help');
-
-    const files = getRemainingArgs();
-    if((!scanStudies && !files.length) || isHelp) {
-        showHelp( 
-            defaults.helpShort || 'mkdicomweb (options) <inputfiles>',
-            defaults.helpLong || 
-            'Make DICOMweb query and metadata from binary Part 10 DICOM files.  Does a full read of\n'+
-            'deduplicated files each time a study instance UID is found, and only updates those studies\n'+
-            'having at least one ');
-        return
+        // resolve promise with statistics
+        return {}
     }
-    
-    const options = {
-        maximumInlinePublicLength, maximumInlinePrivateLength,
-        isGroup, isInstanceMetadata, isDeduplicate,
-        isStudyData, isClean,
-        recompressType, contentType, colourContentType,
-        directoryName, 
-        deduplicatedRoot,
-        deduplicatedInstancesRoot, 
-        removeDeduplicatedInstances,
-        scanStudies,
-        verbose,
+
+
+    /**
+     * The mkdicomweb command first runs mkdicomwebinstances, writing out the deduplicated data, and then runs the
+     * mkdicomwebstudy command, creating the deduplicated data set.  This version, however, keeps the deduplicated
+     * data in memory by default on a study level, which avoids needing to run the load process.
+     */
+    async main() {
+        if (this.options.scanStudies) {
+            // Scan one of the study directories - in this case, files is a set of study directories
+            await this.processStudyDir(this.files, this.options);
+        } else {
+            await this.processFiles(this.files, this.options);
+        }
+        await this.close();
     }
 
-    const callback = {
-        uids: IdCreator(options),
-        bulkdata: HashDataWriter(options),
-        imageFrame: ImageFrameWriter(options),
-        videoWriter: VideoWriter(options),
-        completeStudy: CompleteStudyWriter(options),
-        metadata: InstanceDeduplicate(options),
-        deduplicated: DeduplicateWriter(options),
-        scanStudy: ScanStudy(options),
+    async close() {
+        await this.callback.completeStudy();
+        OverallStats.summarize('Completed Study Processing');
     }
 
-    if( options.scanStudies ) {
-      // Scan one of the study directories - in this case, files is a set of study directories
-      await processStudyDir(files, callback, options);
-    } else {
-      await processFiles(files, callback, options);
+    static main(defaults) {
+        const importer = new StaticWado(defaults);
+        return importer.main();
     }
-    await callback.completeStudy();
-    OverallStats.summarize('Completed Study Processing');
 }
 
-dicomp10todicomweb.main = main;
+new StaticWado({});
```

Comment:
is this line needed?

Reply:
No, was having failures on create in an odd spot, and wanted ot test the constructor.  Forgot to remote it.